### ISSUE TITLE
Sanity check: SFT Model should be frozen (PPO)

### ISSUE THREAD

**Issue by Apsod** [CarperAI/trlx#517]
### 🐛 Describe the bug

When training a Hydra model, the SFT model used for the reference distribution is not fully frozen due to the fact that the embedding layers are left trainable: 

```
        v------------------  WEIGHT TIED  ---------------v
Tokens ~~> Embeddings ==> Hidden state ==> Output state ~~> Token Distribution (SFT)
                                      \
                                       ~~> Output state ~~> Token Distribution (Policy)
                                                       \
                                                        ~~> Value              (Value Head)
~~> Not frozen
==> Frozen
```

# Reproduction

I made a fork with a metric function that uses the reference model to compute the sum of logits for a static dummy input: https://github.com/Apsod/trlx/commit/a7e6fc6fd975c6f51597ad2eea19fca6d51172e5 

Since the SFT-model should be frozen, this reference logit sum should be constant (or vary slightly due to dropout), but this is not the case when running the examples/ppo_sentiments.py-script. I then explicitly froze the embedding layers in `freeze_bottom_causal_layers`, which resulted in constant reference logit sum https://github.com/Apsod/trlx/commit/7c91b450441e19d518e6afee56753ff241ece9a8

![image](https://github.com/CarperAI/trlx/assets/5305850/e4867cc3-d317-4bf2-86e1-6a801adf30fc)

### Which trlX version are you using?

Commit hash: 9c83ceeba0fed7fe856bf92ac02fe5d9957f8f33

**Comment by Dahoas** [comment]
@cat-state Have you gotten a chance to look at this yet?

**Comment by cat-state** [comment]
Thanks, this should be fixed by: https://github.com/CarperAI/trlx/pull/543
